src/App.tsx

```diff
@@ -14,11 +14,101 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[]
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  getGoodsTitle() {
+    if (this.state.selectedGoods.length === 0) {
+      return 'No goods selected';
+    }
+
+    if (this.state.selectedGoods.length === 1) {
+      return `${this.state.selectedGoods[0]} is selected`;
+    }
+
+    return `${this.state.selectedGoods.slice(0, -1).join(', ')
+    } and ${this.state.selectedGoods.slice(-1)} are selected`;
+  }
+
+  render() {
+    return (
+      <div className="
+        container
+        is-max-desktop
+        is-flex
+        is-flex-direction-column
+        is-align-items-center
+        notification
+        is-light
+      "
+      >
+        <progress
+          className="progress is-info"
+          value={this.state.selectedGoods.length}
+          max={goodsFromServer.length}
+        />
+
+        <h1 className="title is-3 has-text-centered">{this.getGoodsTitle()}</h1>
+
+        <ul>
+          {goodsFromServer.map(good => (
+            <li
+              className={`
+                box
+                is-flex
+                is-justify-content-space-between
+                is-align-items-center
+                mb-3
+                ${this.state.selectedGoods.includes(good) && 'has-background-primary'}
```

Comment:
you repeat this `this.state.selectedGoods.includes(good)` check 4 times, move that to variable

Reply:
fixed